Transition config support in StackNavigator

danielkermode opened the pull request:
**Motivation**

I would like a way to adjust the default transition behaviour of `StackNavigator`. I noticed that this change seems to be in the works with #124. My change is a very simple one that allows `transitionConfig` to be passed to `StackNavigator`, giving complete control over `transitionSpec` and `screenInterpolator`. There is probably a far nicer API that could be used, but this one is functional for now.

Added documentation for the option in `StackNavigator.md`.

matthamil commented on docs/api/navigators/StackNavigator.md:
````diff
@@ -128,6 +128,22 @@ All `navigationOptions` for the `StackNavigator`:
+- `transitionConfig` - a function returning an object that will override default animations in `CardStack`.
+Object has optional keys `transitionSpec` and `screenInterpolator`. Example:
+```js
+transitionConfig: () => ({
````
I think this example should provide full context to `transitionConfig`, i.e. showing an entire example of `navigationOptions`. That is, it should resemble the format in other examples dealing with `navigationOptions` like so:


```jsx
class ProfileScreen extends React.Component {

  static navigationOptions = {

    title: ({ state }) => `${state.params.name}'s Profile!`,

    transitionConfig: () => ({
      transitionSpec: {
        // Default duration is 250 (to attempt to match default native behaviour). Override to reduce this.
        duration: 160,
        // Optional easing and timing keys (these are defaults)
        easing: Easing.inOut(Easing.ease),
        timing: Animated.timing
      },
      // screenInterpolator is a function that takes NavigationSceneRendererProps and determines how to interpolate scenes,
      // by itself returning an object with `opacity` and `transform` keys. For examples, look at   `CardStackStyleInterpolator.js`.
      screenInterpolator: MyScreenInterpolator
    })
  };
  ...
```

danielkermode replied:
Fixed docs so that `transitionConfig` is actually part of `StackNavigatorConfig` and not `navigationOptions`.

matthamil commented on yarn.lock:
```diff
@@ -1,7 +1,5 @@
 # THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Are the changes to the `yarn.lock` necessary for this PR?

danielkermode commented:
Removed yarn lock file changes. Documentation was incorrect, `transitionConfig` is actually part of `StackNavigatorConfig` and not `navigationOptions`, fixed now.

ptc-jhughes commented:
What is the status on this PR? Everything seems finished

matthamil commented:
@ptc-jhughes Waiting on the maintainers to approve and merge.

blizzardzheng commented:
so where is the maintainers?

nihgwu commented:
https://github.com/react-community/react-navigation/issues/700#issuecomment-287102525

ptc-jhughes commented:
@nihgwu Agreed. It's not a good practice to let completed branches sit there and run the risk of getting stale relative to master. I fully understand the need for solid code, but the bottleneck for an issue should be creating that code, not waiting for the review process once implemented. If there is anything we can do to help, please advise.

zeh commented:
Nice to know this is moving, I've actually been using this for a while on a local fork and it's working well. Thanks @danielkermode.

grabbou commented:
Closing in favour of #1187. Apologies that it took so long to land.
